spec.bs

```diff
@@ -1148,8 +1148,36 @@ To <dfn>validate and convert auction ad config</dfn> given an {{AuctionAdConfig}
 1. If |config|["{{AuctionAdConfig/serverResponse}}"] [=map/exists=]:
   1. If |config|["{{AuctionAdConfig/requestId}}"] does not [=map/exist=], then [=exception/throw=]
     a {{TypeError}}.
-  1. Set |auctionConfig|'s [=auction config/server response=] to |config|["{{AuctionAdConfig/serverResponse}}"].
-  1. Set |auctionConfig|'s [=auction config/server response id=] to |config|["{{AuctionAdConfig/requestId}}"].
+  1. [=Handle a server response promise in configuration=] given |auctionConfig| and
+    |config|["{{AuctionAdConfig/serverResponse}}"]:
+
+    Note: The JavaScript code calling {{Navigator/runAdAuction()}} is responsible for *not*
+      resolving |config|["{{AuctionAdConfig/serverResponse}}"] until server responses have been
+      retrieved from one or more [:Ad-Auction-Result:] headers, as resolving this Promise early
+      would cause a race condition in which the [=script runners=] might run without the server
+      responses that it needs. There are two ways that server responses can be retrieved.
+      (1) The JavaScript code issues a [=request=] whose [=request/initiator type=]
+      is `"fetch"` and whose {{RequestInit/adAuctionHeaders}} option is set to `true`. The
+      JavaScript code has to resolve |config|["{{AuctionAdConfig/serverResponse}}"] after the
+      corresponding call to {{WindowOrWorkerGlobalScope/fetch()}} has resolved to a [=response=].
+      The JavaScript code can also choose to wait to call {{Navigator/runAdAuction()}} until after
+      the corresponding call to {{WindowOrWorkerGlobalScope/fetch()}} has resolved to a [=response=],
+      and can then immediately resolve |config|["{{AuctionAdConfig/serverResponse}}"];
+      (2) issue an <a spec="html" lt="navigate an iframe or frame">iframe navigation</a>
+      request with the <{iframe/adauctionheaders}> <a spec=html>content attribute</a> set to `true`.
+      In this case, the JavaScript code is retrieved as part of the iframe navigation response,
+      at which point the JavaScript code in the iframe makes the call to
+      {{Navigator/runAdAuction()}}, and |config|["{{AuctionAdConfig/serverResponse}}"] can be
+      specified directly without a Promise.
+
+    * To parse the value |result|:
+      1. Set |auctionConfig|'s [=auction config/server response=] to

```

Comment:
check if this struct's field can be uint8array, or needs another Infra data structure, like byte sequence. If so, need to do convert javascript to infra value on it here.

Reply:
I think it's okay to leave it as a Uint8Array, but we probably need to copy or transfer (https://webidl.spec.whatwg.org/#arraybuffer-transfer) the underlying storage. UInt8Array is really just a shallow pointer to a buffer. We're going to be accessing it [=in parallel=] and we want to specify that later changes to the provided array in the main context won't affect the version we are using in the parallel context.

Reply:
how about this https://webidl.spec.whatwg.org/#dfn-get-buffer-source-copy? Does it match our intention?

Reply:
Yes, I would prefer a transfer here over a copy. With a transfer the value in the array is undefined after a call, but allows implementations to optimize. We perform a copy on Chrome because of IPC, but there's no reason to require a copy.

Reply:
got it. Done.